### ISSUE TITLE
Parsing long expressions creates infinite recursion

### ISSUE THREAD

**Issue by mgurmendez** [bpodgursky/jbool_expressions#6]
This test reproduces the issue:
```
public void testLongExpression() {
      String exp = "(122036 | 122037 | 122039 | 122040 | 122042 | 122043 | 122045 | 122046 | 122048 | 122049 | " +
              "122051 | 122052 | 122054 | 122055 | 122057 | 122058 | 122060 | 122061 | 122063 | 122064 | 122066 | " +
              "122067 | 122069 | 122070 | 169225 | 169226 | 169228 | 169229 | 169231 | 169232 | 169234 | 169235 | " +
              "169237 | 169238 | 169240 | 169241 | 169243 | 169244 | 169246 | 169247 | 169249 | 169250 | 169252 | " +
              "169253 | 169255 | 169256 | 169258 | 169259 | 169261 | 169262 | 169264 | 169265 | 106925 | 106926 | " +
              "165611)";
      ExprParser.parse(exp);
}
```

I fixed this by caching the string representation in the Or, And and Not operations. If you're interested I can share the PR. 

PD Great library, by the way! Thanks for sharing as open source!

**Comment by bpodgursky** [comment]
If you have a PR that fixes it I'd be happy to merge it.

**Comment by bpodgursky** [comment]
I just released v1.7 with those changes -- should show up in maven central sometime later today.  Thanks for the fixes, glad you found it useful!
